### ISSUE TITLE
Adding a track without playing it

### ISSUE THREAD

**Issue by ataravati** [react-native-kit/react-native-track-player#404]
Is there any way to add a track to the queue without playing it? What I want to achieve is adding tracks one at a time to the queue and then playing them only when the play method is called.

**Comment by akomkov** [comment]
Same question here

**Comment by minhtc** [comment]
store all yours tracks in a variable (e.g: ```const pending Tracks =[]``` ), then add to TrackPlayer when you actually need to play it.

**Comment by ataravati** [comment]
@minhtc That's not the point. The point is when you add a track, it shouldn't be automatically played. Adding a track to the player should only load the track and change the status to ready (or whatever the status should be). This is an obvious violation of SoC. Every function should be doing one and only one thing.

**Comment by Guichaguri** [comment]
Which platform are you facing this issue?

**Comment by ataravati** [comment]
@Guichaguri I'm having the issue in iOS. I haven't tried it in Android though. So, is this a bug or is it by design?

**Comment by minhtc** [comment]
apply this commit will fix your issue: https://github.com/HomieTech/react-native-track-player/commit/49c53f88270273c027d225e9a0c78dc149e8c11f

**Comment by dcvz** [comment]
@Guichaguri I believe on Android we also have it to where adding the first track automatically plays it, no?

**Comment by Guichaguri** [comment]
@dcvz In Android, it's a boolean property that defines if it will play or not that can be changed through `play()` or `pause()`

```js
TrackPlayer.play(); // sets "play when ready" to true
TrackPlayer.add(...); // as soon as the track loads, it will start playing
```

```js
TrackPlayer.pause(); // sets "play when ready" to false
TrackPlayer.add(...); // it will load the track, but not start it yet
```

**Comment by ataravati** [comment]
@Guichaguri that's exactly what I believe should change. `TrackPlayer.add(...);` should only add the track and set the status to "ready".

**Comment by dcvz** [comment]
I will fix this shortly 💃 I'll keep ya'll posted.

**Comment by dcvz** [comment]
I'm going to set it to "adding a track when there's no tracks in queue will not automatically start". You'll have to then call `play()`. /cc @Guichaguri

**Comment by dcvz** [comment]
This is now fixed in: 31fd540278e463d0b9051fcc09886b71dd26ed8c

**Comment by ataravati** [comment]
It works now. Thanks!

**Comment by curiousdustin** [comment]
This also fixes some strange behavior if you are trying to start a track from a certain point such as:
```
await TrackPlayer.reset();
await TrackPlayer.add(track);
await TrackPlayer.seekTo(position);
await TrackPlayer.play();
```
Without this fix, sometimes this ends up with the track paused at the position, instead of playing.

**Comment by CapitanRedBeard** [comment]
For anyone wondering this landed right after 1.1.2 so presumably it will be ready in 1.1.3, although you can access it now by pointing to the `dev` branch

**Comment by vlatri** [comment]
Hey, @dcvz it seems to me that `TrackPlayer.add` still triggers `playback-track-changed` event with `nextTrack` containing the id of the very first queue item. Is this still intended? Shouldn't it only get fired on `play` after adding tracks?
